### PR TITLE
ci: Improve test coverage

### PR DESCRIPTION
**Pull request by dplewis**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-JS/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-JS/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).

## Issue

Add missing coverage

## Approach

* Test unsupported live queries
* Test destroy with session token
* Test live query connections / disconnection
* Test internal nested object states

- [x] Add tests

### CODE REVIEW

File: src/OfflineQuery.js [hunk 1]
```diff
@@ -538,7 +538,7 @@ function matchesKeyConstraints(className, object, objects, key, constraints) {
         const distance = point.radiansTo(centerPoint);
         return distance <= maxDistance;
       }
-      break;
+      return false;
```
Comment (mtrezza) [inline src/OfflineQuery.js:541]: Why this change? Seems to go beyond "test coverage" - is this a bug fix?

Comment (dplewis) [reply src/OfflineQuery.js:541]: It’s technically unreachable code. The change is for the test to see what happens if it is reached

Comment (mtrezza) [reply src/OfflineQuery.js:541]: Are you sure it's unreachable? I haven't looked much into this, but what if the constraint is neither `$polygon` nor `$centerSphere`, wouldn't it be reachable?

Comment (dplewis) [reply src/OfflineQuery.js:541]: Using the SDK as is it’s unreachable. You could mess around with the internals to reach it like I do with the tests

Comment (mtrezza) [reply src/OfflineQuery.js:541]: Got it, and have you looked into what the difference is between `break` and `return false`? For example `case '$maxDistance'` breaks, it doesn't return false, so maybe there is a reason why this here has a `break` and is not returning false - even if it's (currently) unreachable?

Comment (dplewis) [reply src/OfflineQuery.js:541]: $maxDistance is ignored so the break statement falls to true (matching) ignoring the default (false). Here in $geoWithin this would be non matching invalid query and should return false instead of break which would be true.

It has a break statement because a test wasn’t written for it and was overlooked

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.2.0-alpha.7](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.2.0-alpha.7)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.0-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.0-alpha.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.0)
